packages/caver-account/src/accountKey/accountKeyRoleBased.js

```diff
@@ -0,0 +1,149 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const RLP = require('eth-lib/lib/rlp')
+const AccountKeyWeightedMultiSig = require('./accountKeyWeightedMultiSig')
+const AccountKeyLegacy = require('./accountKeyLegacy')
+const AccountKeyPublic = require('./accountKeyPublic')
+const AccountKeyFail = require('./accountKeyFail')
+const utils = require('../../../caver-utils')
+const { ACCOUNT_KEY_TAG } = require('./accountKeyHelper')
+const { KEY_ROLE } = require('../../../caver-keyring/src/keyringHelper/keyringHelper')
+
+function isArrayOfAccountKeyFormat(roleBasedAccountKeys) {
```

Comment:
What do you think of the function name like "isValidRoleBasedKeyFormat" or something similar?

Reply:
Okay, sounds good :) Thank you for suggestion.